Configuration.cpp

```diff
@@ -100,33 +101,42 @@ bool Configuration::readConfigFile(char *path) {
   if(all_udp_ports) trace->traceEvent(TRACE_INFO, "All UDP ports will be monitored");
 
 
-  int counter = 2;
-  do {
-  std::string json_value_str = counter == 2 ? "countries" : "continents";
-  if(!root[json_value_str].empty()) {
-    if(root[json_value_str]["whitelist"].empty()) {
-      trace->traceEvent(TRACE_INFO, "Missing %s from %s", "whitelist", path);
-    } else {
-      for(Json::Value::ArrayIndex i = 0; i != root[json_value_str]["whitelist"].size(); i++) {
-	std::string ctry_cont = root[json_value_str]["whitelist"][i].asString();
-
-	trace->traceEvent(TRACE_INFO, "Adding %s to %s whitelist", ctry_cont.c_str(), json_value_str.c_str());
-	ctrs_conts[ctry_cont2u16((char*)ctry_cont.c_str())] = MARKER_PASS;
-      }
-    }
-
-    if(root[json_value_str]["blacklist"].empty()) {
-      trace->traceEvent(TRACE_INFO, "Missing %s from %s", "blacklist", path);
-    } else {
-      for(Json::Value::ArrayIndex i = 0; i != root[json_value_str]["blacklist"].size(); i++) {
-	std::string ctry_cont = root[json_value_str]["blacklist"][i].asString();
-
-	trace->traceEvent(TRACE_INFO, "Adding %s to %s blacklist", ctry_cont.c_str(), json_value_str.c_str());
-	ctrs_conts[ctry_cont2u16((char*)ctry_cont.c_str())] = MARKER_DROP;
-      }
-    }
+  std::string json_policy_str = default_policy == MARKER_DROP ? "drop" : "pass";
+
+  if(!root["policy"].empty() && !root["policy"][json_policy_str].empty()) {
+
+      Json::Value json_policy_obj = root["policy"][json_policy_str];
+      int counter = 2;
+      do {
+        std::string json_value_str = counter == 2 ? "countries" : "continents";
+        
+        if(json_policy_str == "drop") { /* parsing only whitelists */
+          
+          if(json_policy_obj[json_value_str+"_whitelist"].empty()) {
+            trace->traceEvent(TRACE_INFO, "Missing %s whitelist from %s", json_value_str.c_str(), path);
+          } else {
+            for(Json::Value::ArrayIndex i = 0; i != json_policy_obj[json_value_str+"_whitelist"].size(); i++) {
+        std::string ctry_cont = json_policy_obj[json_value_str+"_whitelist"][i].asString();
+
+        trace->traceEvent(TRACE_INFO, "Adding %s to %s whitelist", ctry_cont.c_str(), json_value_str.c_str());
+        ctrs_conts[ctry_cont2u16((char*)ctry_cont.c_str())] = MARKER_PASS;
+            }
+          }
+        } else { /* parsing only blacklists */
```

Comment:
Can you avoid this else branch and use a `str::string json_list_str = json_policy_str == "drop" ? "_whitelist" : "_blacklist"`? Something similar to what you did with `json_value_str`

Reply:
Right, that else branch was absolutely avoidable, thanks!